src/model.py

```diff
@@ -2106,7 +2242,9 @@ def forward(self, batch):
                 else:
                     same_task = [attn_inter]
 
-                task_inputs[first_task] = torch.stack(same_task).mean(dim=0)  
+                if first_task in task_inputs:
+                    re_encoded = self.components[first_task]["re_encode"](torch.stack(same_task).mean(dim=0))
+                    task_inputs[first_task] = re_encoded.permute(0, 2, 1)
```

Comment:
Since the attention output is now used as the next task_input, we need to re-encode it to the orignal shared embedding size